src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,72 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
-    private static final Injector injector = new Injector();
+    private static final Injector INJECTOR = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>(
+            Map.of(
+                    FileReaderServiceImpl.class, new FileReaderServiceImpl(),
+                    ProductParserImpl.class, new ProductParserImpl(),
+                    ProductServiceImpl.class, new ProductServiceImpl()
+            )
+    );
 
     public static Injector getInjector() {
-        return injector;
+        return INJECTOR;
+    }
+
+    public <T> T getInstance(Class<T> interfaceClass) {
+        if (instances.containsKey(interfaceClass)) {
+            return interfaceClass.cast(instances.get(interfaceClass));
```

Comment:
```suggestion
            return instances.get(interfaceClass);
```
by interface you get an implementation, no need to cast anything

Reply:
![image](https://github.com/user-attachments/assets/3a67840c-d3c0-448d-9f4e-a7e0d4f045fd)